sdk/go/pulumi/resource.go

```diff
@@ -369,6 +379,12 @@ type ResourceOptions struct {
 	// the resource's properties during construction.
 	Transformations []ResourceTransformation
 
+	// XTransforms is a list of functions that transform
+	// the resource's properties during construction.
+	//
+	// Experimental.
```

Comment:
Is it worth explaining a bit more what "Experimental" entails?  E.g. API changes are expected, behaviour changes are expected, etc.

Reply:
We should probably just have a doc somewhere on the site to explain it and link to that, rather than repeating it everywhere. But I imagine most people are ok with what experimental means.